### PR TITLE
fby3.5: cl: Support to get DIMM I3C mux selection by IPMI command

### PR DESCRIPTION
**Pull request by DelphineChiu**
Summary:
- Create an new IPMI command "OEM_1S_GET_DIMM_I3C_MUX_SELECTION" for BIOS to get the dimm mux selection (BIC or CPU).

Dependency: #801 

Test plan:
- Build code: Pass
- BMC to BIC through IPMB : Pass
- Host to BIC through KCS : Pass

Log:
1. BMC to BIC through IPMB:
   root@bmc-oob:~# bic-util 0xe0 0xb2 0x15 0xa0 0x00
   15 a0 00 01

2. Host to BIC through KCS:
   [root@FBK8_221107 ~]# ipmitool raw 0x38 0xb2 0x15 0xa0 0x00
   15 a0 00 01

### CODE REVIEW

**Comment by DelphineChiu** [comment]
Dependency: #801

File: meta-facebook/yv35-cl/src/platform/plat_hook.c [hunk 1]
```diff
@@ -405,3 +406,39 @@ bool post_ltc4282_read(uint8_t sensor_num, void *args, int *reading)
...
+	dimm_pre_proc_arg *pre_proc_args = (dimm_pre_proc_arg *)args;
+	if (pre_proc_args->is_present_checked == true) {
+		return true;
+	}
+
+	uint8_t dimm_id = -1;
+	bool ret = false;
+
+	dimm_id = sensor_num_map_dimm_id(sensor_num);
+	if (dimm_id < 0) {
```
Comment (garnermic) [inline meta-facebook/yv35-cl/src/platform/plat_hook.c:428]: Check of uint8_t for < 0.

**Review by garnermic — CHANGES_REQUESTED** [review]
Changes requested based on Meta review.

**Comment by SaraLin-wiwynn** [comment]
Hi @garnermic 
Thanks for your comments.
Based on your suggestions to modify the code and force update this branch.
Please help review it again, thank you.

File: common/dev/i3c_dimm.c [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	int ret = 0;
+	uint8_t data[2] = { 0 };
+
+	memset(data, 0, sizeof(data));
+	memset(reading, 0, sizeof(int));
+	sensor_val *sval = (sensor_val *)reading;
+
+	switch (sensor_config[sensor_config_index_map[sensor_num]].offset) {
+	case DIMM_PMIC_SWA_PWR:
+		ret = pal_get_pmic_pwr(sensor_num, (uint8_t *)&data);
```
[the remark this answers is not in the document]
Comment (SaraLin-wiwynn) [reply common/dev/i3c_dimm.c]: Correct it, thank you.

File: meta-facebook/yv35-cl/src/platform/plat_dimm.c [hunk 3]
```diff
@@ -0,0 +1,447 @@
...
+		switch_i3c_dimm_mux(I3C_MUX_TO_BIC, DIMM_MUX_TO_DIMM_A0A1A3);
+
+		if (k_mutex_unlock(&i3c_dimm_mux_mutex)) {
+			LOG_ERR("Failed to unlock I3C dimm MUX");
+		}
+	}
+
+	while (1) {
+		int dimm_id = 0, ret = 0;
+		I3C_MSG i3c_msg = { 0 };
```
[the remark this answers is not in the document]
Comment (SaraLin-wiwynn) [reply meta-facebook/yv35-cl/src/platform/plat_dimm.c]: Remove it, thank you.

File: meta-facebook/yv35-cl/src/platform/plat_hook.c [hunk 4]
```diff
@@ -405,3 +406,39 @@ bool post_ltc4282_read(uint8_t sensor_num, void *args, int *reading)
...
+		// BIC can't check DIMM present status, return true to keep do sensor initial
+		return true;
+	}
+
+	dimm_pre_proc_arg *pre_proc_args = (dimm_pre_proc_arg *)args;
+	if (pre_proc_args->is_present_checked == true) {
+		return true;
+	}
+
+	uint8_t dimm_id = -1;
```
[the remark this answers is not in the document]
Comment (SaraLin-wiwynn) [reply meta-facebook/yv35-cl/src/platform/plat_hook.c]: We add a new definition "DIMM_ID_UNKNOWN" to check if dimm_id can't map.

File: meta-facebook/yv35-cl/src/platform/plat_hook.c (hunk 1 above)
Comment (SaraLin-wiwynn) [reply meta-facebook/yv35-cl/src/platform/plat_hook.c]: We add a new definition "DIMM_ID_UNKNOWN" to check if dimm_id can't map.

**Comment by garnermic** [comment]
Can you check this PR and rebase it as the changes seem to conflict with PR #803?

Thanks!

**Comment by SaraLin-wiwynn** [comment]
Hi @garnermic ,
We rebase PR#803 and force update this branch, thank you.

**Pull request closed without merging**